packages/react/src/table/__tests__/index.test.tsx

```diff
@@ -351,4 +353,69 @@ describe('Table', () => {
     expect(thirdColumnHeader.text()).toBe('DATE MODIFIED');
     expect(thirdColumnHeader.props()['aria-sort']).toBe('none');
   });
+
+  it('should reset pagination when amount of rows change', async () => {
```

Comment:
@jrgarciadev probably this does not look great to you and I apologise, I'm not very familiar with enzyme. What I try to check here is that indeed the pagination reset when there is an items update.